src/include/utils/palloc_memory_debug.h

```diff
@@ -0,0 +1,80 @@
+#ifndef PALLOC_OVERRIDE_H
+#define PALLOC_OVERRIDE_H
+
+#define MAX_TOP_ALLOC_CHUNK_STATS 10
+#define CHUNKS_TABLE_SIZE 1024
+
+typedef struct
+{
+	uint64_t count;
+	uint64_t bytes;
+} MemoryContextChunkStat;
+
+typedef struct
+{
+	const char *parent_func;
+	int line;
+} MemoryContextChunkStatKey;
+
+typedef struct
+{
+	#define DYNAMIC_MEMORY_DEBUG_INIT_MAGIC 0x12345678
+	int32_t init;
+	MemoryContextChunkStatKey key;
+	const char *file;
+	const char *exec_func;
+} MemoryContextChunkInfo;
+
+typedef struct
+{
+	MemoryContextChunkStatKey key;
```

Comment:
I'm sorry,  but can you remind me a talk about this structure member? MemoryContextChunkInfo already has a key, but you mentioned some bugs around. Did you try to switch `key` and `init` members of MemoryContextChunkInfo, so we can avoid key duplication?